### PR TITLE
Use retry action for prober

### PR DESCRIPTION
**Pull request by malancas**
#### Summary

Closes https://github.com/sigstore/public-good-instance/issues/1113

Let's try using a [Github Action for retrying a specific job](https://github.com/marketplace/actions/retry-step) to hopefully address some occasionally flakiness we see in the GHA environment. Occasionally the `sigstore-prober` job step will fail because it cannot find a variable that should be available in the GHA environment. The job runs as expected when we manually retry it.

By allowing retries on the sigstore-prober job, we can hopefully avoid alerting on this occasional flakiness and only alert on repeated failures.

### CODE REVIEW

File: .github/workflows/reusable-prober.yml [hunk 1]
```diff
@@ -87,7 +87,12 @@ jobs:
       - name: Run prober
         env:
           DEBUG: 1
-        run: prober --one-time --rekor-url ${{ inputs.rekor_url }} --fulcio-url ${{ inputs.fulcio_url }}
+        uses: nick-fields/retry@943e742917ac94714d2f408a0e8320f2d1fcafcd # v2.8.3
```
Comment (malancas) [inline .github/workflows/reusable-prober.yml:90]: If we want to try running this only in staging to start, we could add a new job step that uses the `retry` action and leave this one untouched. We could then use an input to signify whether we want to use the existing `Run prober` job step or the new experimental one.

**Review by cpanato — APPROVED** [review]
lgtm

File: .github/workflows/reusable-prober.yml [hunk 2]
```diff
@@ -87,7 +87,13 @@ jobs:
...
+        with:
+          timeout_minutes: 5
+          max_attempts: 3
+          retry_wait_seconds: 10
```
Comment (haydentherapper) [inline .github/workflows/reusable-prober.yml:94]: Maybe a minute? This seems small

File: .github/workflows/reusable-prober.yml [hunk 3]
```diff
@@ -87,7 +87,13 @@ jobs:
       - name: Run prober
         env:
           DEBUG: 1
-        run: prober --one-time --rekor-url ${{ inputs.rekor_url }} --fulcio-url ${{ inputs.fulcio_url }}
+        uses: nick-fields/retry@943e742917ac94714d2f408a0e8320f2d1fcafcd # v2.8.3
+        with:
+          timeout_minutes: 5
```
Comment (haydentherapper) [inline .github/workflows/reusable-prober.yml:92]: What’s the average length of a run? Is 5 too long?

**Review by vaikas — DISMISSED** [review]
Thank you for doing this!! 🤞

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**